src/Adapter/LocalAdapter.php

```diff
@@ -460,4 +460,43 @@ private function deleteDirByFullPath($path, $delete_self = true, array $exclude
             rmdir($dir);
         }
     }
-}
\ No newline at end of file
+
+    /**
+     * {@inheritdoc}
+     */
+    public function changePermissions($path, $mode = 0777)
```

Comment:
For recursive changePermissions try something like this:

```php
$iterator = new RecursiveIteratorIterator(new RecursiveDirectoryIterator($pathname));

foreach($iterator as $item) {
    chmod($item, $filemode);
}
```

Taken from PHP docs (comments).